fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/ElectricVehicleUnit.java

```diff
@@ -45,54 +45,58 @@ public ElectricVehicleUnit(String vehicleName, VehicleType vehicleType, final Ge
         setRequiredOperatingSystem(ElectricVehicleOperatingSystem.class);
     }
 
-    private void updateBatteryInformation(final VehicleBatteryState currentVehicleBatteryState) {
+    private void updateBatteryInformation(final BatteryData currentBatteryData) {
         // set the new vehicle electric info reference
-        VehicleBatteryState previousVehicleBatteryState = this.vehicleBatteryState;
-        this.vehicleBatteryState = currentVehicleBatteryState;
+        BatteryData previousBatteryData = this.batteryData;
+        this.batteryData = currentBatteryData;
 
         for (ElectricVehicleApplication application : getApplicationsIterator(ElectricVehicleApplication.class)) {
-            application.onBatteryStateUpdated(previousVehicleBatteryState, currentVehicleBatteryState);
+            application.onBatteryDataUpdated(previousBatteryData, currentBatteryData);
         }
     }
 
-    private void onVehicleChargingDenialResponse(final ChargingDenialResponse chargingDenialResponse) {
+    private void onVehicleChargingDenialResponse(final VehicleChargingDenial vehicleChargingDenial) {
         for (ElectricVehicleApplication application : getApplicationsIterator(ElectricVehicleApplication.class)) {
-            application.onChargingRequestRejected(chargingDenialResponse);
+            application.onChargingRequestRejected(vehicleChargingDenial);
         }
     }
 
     @Override
     protected boolean handleEventResource(Object resource, long eventType) {
-        if (resource instanceof VehicleBatteryState) {
-            updateBatteryInformation((VehicleBatteryState) resource);
+        if (resource instanceof BatteryData) {
+            updateBatteryInformation((BatteryData) resource);
             return true;
         }
 
-        if (resource instanceof ChargingDenialResponse) {
-            onVehicleChargingDenialResponse((ChargingDenialResponse) resource);
+        if (resource instanceof VehicleChargingDenial) {
+            onVehicleChargingDenialResponse((VehicleChargingDenial) resource);
```

Comment:
The order of methods is weird here. Usually, private method declarations should be placed below the line where they are called from, not above. Just put `handleEventResource` above `updateBatteryInformation`